src/components/TodoItem.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../types/Todo';
+import { TodosContext } from '../hooks/TodosContext';
+import { useLocalStorage } from '../hooks/LocalStorage';
+
+type TodoInfoProps = {
+  todo: Todo
+};
+
+export const TodoItem = ({ todo: { id, title, completed } }: TodoInfoProps) => {
+  const [editedId, setEditedId] = useState(-1);
+  const [editedTitle, setEdtedTitle] = useState(title);
+  const { todos, setNewTodos } = useContext(TodosContext);
+  const [, setStorageTodos] = useLocalStorage();
+  const editedTitleInput = useRef<HTMLInputElement>(null);
+
+  const itemClass = classNames({ completed }, {
+    editing: id === editedId,
+  });
+
+  const handleLabelChange = () => {
+    const newTodos = todos.map(todo => {
+      const todoCopy = { ...todo };
+
+      if (todo.id === id) {
+        todoCopy.completed = !todo.completed;
+      }
+
+      return todoCopy;
+    });
+
+    setNewTodos(newTodos);
+    setStorageTodos(newTodos);
+  };
+
+  const handleClick = () => {
+    const newTodos = todos.filter(todo => todo.id !== id);
+
+    setNewTodos(newTodos);
+    setStorageTodos(newTodos);
+  };
+
+  const handleDoubleClick = (event: React.MouseEvent) => {
+    if (event.detail === 2) {
```

Comment:
pls no magic numbers, move `2` to a const with an appropriate name
**DO THE SAME FOR ALL MAGIC NUMBERS IN UR PROJECT**

Reply:
I extracted constants.